compiler/src/scala/quoted/runtime/impl/printers/SourceCode.scala

```diff
@@ -1368,28 +1368,24 @@ object SourceCode {
 
     private def printProtectedOrPrivate(definition: Definition): Boolean = {
       var prefixWasPrinted = false
-      def printWithin(within: TypeRepr) = within match {
-        case TypeRef(_, name) => this += name
-        case _ => printFullClassName(within)
-      }
-      if (definition.symbol.flags.is(Flags.Protected)) {
+      def printWithin(within: Option[TypeRepr]) = within match
```

Comment:
We should have a test for this change.

Reply:
I added a test case for printing the modifiers to the second commit.